src/ext/database_diagram.py

```diff
@@ -0,0 +1,188 @@
+         CylcSuiteDAO.TABLE_TASK_OUTPUTS],
+        [CylcSuiteDAO.TABLE_TASK_POOL, ONE_TO_ONE,
+         CylcSuiteDAO.TABLE_TASK_TIMEOUT_TIMERS]
+    ]
+
+
+def get_columns_metadata(table_name, conn):
+    """Return the metadata for the columns in a table.
+
+    Args:
+        table_name (str): name of the DB table to retrieve the columns.
+        conn (sqlite3.Connection): SQLite connection.
+    Returns:
+        list: an array with the cid, name, type, notnull, default_value,
+        and pk value (0=not, 1=yes).
+    """
+    with closing(conn.cursor()) as cursor:
+        cursor.execute(f'PRAGMA table_info({table_name})')
+        return cursor.fetchall()
+
+
+def schema_to_markdown(db_name):
+    """Return the database markdown schema.
+
+    Args:
+        db_name (str): database name
+    Returns:
+        tuple: one side with lines of a markdown file, the other with
+            the orphan tables
+    """
+    lines = []
+    orphans = []
+    relationships = get_relationships()
+    with closing(sqlite3.connect(db_name)) as conn:
+        with closing(conn.cursor()) as cursor:
+            cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
+            for row in cursor:
+                table = row[0]
+                lines.append(f"[{table}]")
+                for column in get_columns_metadata(table, conn):
+                    pk = "*" if column[5] > 0 else ""  # 1 is normally PK
+                    label = f"{{label: {column[2].upper()}}}"
+                    line = f"\t{pk}{column[1]} {label}"
+                    lines.append(line)
+
+                if not any((
+                    table in relationship
+                    for relationship in relationships
+                )):
+                    orphans.append(table)
+
+            # in the eralchemy example, relationships go at the end of the file
+            lines.extend([" ".join(value) for value in relationships])
+            lines.extend(orphans)
+    return lines, orphans
+
+
+def group_nodes(nodes):
+    """Place the provided nodes in a subgraph to keep them together."""
+    lines = [
+        'subgraph {',
+        'rank=same',
+        'rankdir=LR'
+    ]
+    lines.extend([f'"{x}"' for x in nodes])
+    return lines + ['}']
+
+
+class CylcRunDBDirective(Graphviz):
+    """Directive which inserts a Cylc run DB graph."""
+
+    option_spec = dict(Graphviz.option_spec)
+    required_arguments = 0
+
+    @staticmethod
+    def generate_dotcode():
+        # get markdown
+        with tempfile.NamedTemporaryFile() as tf_db:
+            # is_public=False triggers the creation of tables
+            CylcSuiteDAO(db_file_name=tf_db.name, is_public=False)
+            schema, orphans = schema_to_markdown(db_name=tf_db.name)
+
+        # graph prefix
+        dotcode = [
+            'graph {',
+            'node [label = "\\N", shape = plaintext];',
+            'edge [color = gray50, minlen = 2, style = dashed];',
+            'rankdir = "TB";'
```

Comment:
We may want to change this to:

```suggestion
            'rankdir = "LR";'
```

That way the graph becomes tall and thin making it large enough to read.

Unfortunately the table relationships don't line up with the field names which makes `LR` nicer (because the arrows end on the table not the table fields).

Reply:
I tried with `TB` and `LR`, but there was no change. Probably because of the table relationships issue you mentioned?

Reply:
Works for me, you will need to `make clean html` when fiddling with scripts/extensions as Sphinx performs an incremental build.

![Screenshot from 2019-09-03 09-36-28](https://user-images.githubusercontent.com/16705946/64157291-5caa6880-ce2e-11e9-884d-3db88d9b87d3.png)

Reply:
Weird. Should we raise a follow up PR to change it?

Reply:
It worked with the `clean` target. Should we change the PR in cylc-doc? I'm fine with either version :)

Reply:
It's kinda nicer the way it is but I don't think anyone will be able to read it (you can't just open the image in a new tab as its an SVG).

Reply:
Good point, took me a few seconds to realize that wasn't an image any more. The good thing about that, is that you can copy the text from the diagram, and there is also a tooltip in the browser, which is nice too.

Pushing the change now. Thanks!